.github/workflows/pr-contract-check.yml

```
@@ -0,0 +1,171 @@
+          # Check and install cargo-contract if not cached
+          if ! command -v cargo-contract &> /dev/null; then
+            cargo install cargo-contract --version 3.2.0
+          fi
+          
+          sudo apt-get update && sudo apt-get install -y python3
+          
+          # Check and install b2sum if not cached
+          if ! command -v b2sum &> /dev/null; then
+            cargo install b2sum
+          fi
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)